polygon/sync/block_downloader.go

```diff
@@ -9,18 +9,23 @@ import (
 	"sync"
 	"time"
 
+	"github.com/c2h5oh/datasize"
 	lru "github.com/hashicorp/golang-lru/v2"
 	"github.com/ledgerwatch/log/v3"
 
 	"github.com/ledgerwatch/erigon-lib/common"
 	"github.com/ledgerwatch/erigon/core/types"
+	"github.com/ledgerwatch/erigon/eth/ethconfig/estimate"
 	"github.com/ledgerwatch/erigon/polygon/heimdall"
 	"github.com/ledgerwatch/erigon/polygon/p2p"
 )
 
 const (
 	blockDownloaderLogPrefix      = "BlockDownloader"
 	notEnoughPeersBackOffDuration = time.Minute
+
+	// conservative over-estimation: 1 MB block size x 1024 blocks per waypoint
```

Comment:
I'm curious if `RLP(body)` is 100 Kb, how big is the Body struct in memory (with all subobjects included)?

Reply:
When block explorers show some "block size" data, it is about the encoded RLP packet size. When we decode it, it will grow in memory, and I wonder by how much. This is just to get some intuition about the memory usage.

Reply:
ah I see ok - I will try to answer this question with some logging and experimentation